game-app/game-core/src/main/java/games/strategy/triplea/delegate/TechTracker.java

```diff
@@ -41,6 +38,10 @@ static class Key {
 
   private final Map<Key, Object> cache = new ConcurrentHashMap<>();
 
+  public void onTechnologyChanged() {
```

Comment:
Consider renaming this method to 'clearCache' so it is a bit more explicit what it is doing. With the current naming, it feels like the method should accept a parameter for the technology that has changed, or a parameter for the new set of technologies, or perhaps some sort of technology object.

Side-comment: It would be nice to have a more automatic way to notify the cache that it needs to be updated, perhaps via an event-listener style pattern (though, I don't really see a super clean way to do this without blowing up the scope of this PR).

Reply:
Renamed, thanks! Agree that I would have preferred if there was a clearer way to listen to technology changes, but unfortunately it's all scattered around like this.